src/main/webapp/app/admin/user-management/user-management.component.ts

```diff
@@ -103,6 +190,187 @@ export class UserManagementComponent implements OnInit, OnDestroy {
         this.dialogErrorSource.unsubscribe();
     }
 
+    /**
+     * Inits the available filter and maps the functions.
+     */
+    initFilters() {
+        this.filters.authorityFilter = this.initFilter(UserStorageKey.AUTHORITY, AuthorityFilter) as Set<AuthorityFilter>;
+        this.filters.originFilter = this.initFilter(UserStorageKey.ORIGIN, OriginFilter) as Set<OriginFilter>;
+        this.filters.statusFilter = this.initFilter(UserStorageKey.STATUS, StatusFilter) as Set<StatusFilter>;
+
+        let key = this.localStorage.retrieve(UserStorageKey.NO_COURSE);
+        this.filters.noCourse = key ? (key as boolean) : false;
+
+        key = this.localStorage.retrieve(UserStorageKey.NO_AUTHORITY);
+        this.filters.noAuthority = key ? (key as boolean) : false;
+    }
+
+    /**
+     * Inits a specific filter.
+     * @param key of the filter in the local storage
+     * @param type of filter
+     */
+    initFilter(key: UserStorageKey, type: any) {
+        const temp = this.localStorage.retrieve(key);
+        const tempInStorage =
+            temp !== undefined && temp !== null
+                ? temp
+                      .split(',')
+                      .map((filter: string) => type[filter])
+                      .filter(Boolean)
+                : new Set();
+        return new Set(tempInStorage);
+    }
```

Comment:
I suggest this to simplify the inner logic a bit, prevent the use of ``any``, and give it a proper return type. There could be better ways to do it than this, but we should avoid using ``any`` everywhere if possible

```suggestion
    initFilter<E>(key: UserStorageKey, type: { [key: string]: string }): Set<E> {
        const temp = this.localStorage.retrieve(key);
        const tempInStorage =
            temp
                ?.split(',')
                .map((filter: string) => type[filter])
                .filter(Boolean) ?? [];
        return new Set(tempInStorage);
    }
```

You can replace the places a few lines above where you use the method like and thereby get rid of the cast:
```js
        this.filters.authorityFilter = this.initFilter<AuthorityFilter>(UserStorageKey.AUTHORITY, AuthorityFilter);
        this.filters.originFilter = this.initFilter<OriginFilter>(UserStorageKey.ORIGIN, OriginFilter);
        this.filters.statusFilter = this.initFilter<StatusFilter>(UserStorageKey.STATUS, StatusFilter);
```

In general, there's a lot of ``any`` below. You could try similar solutions there to get rid of them, e.g. by containing a combined type of your filter enums.

Reply:
Thanks for your suggestions: Solved in cb951a1a5f3c417e2126fb154ff154eab965a1d3 :D